docs/aws/audit/snsmonitoring/rules/is_topic_subscription_public_remediation.mdx

```diff
@@ -1,6 +1,249 @@
+
+4. Check the "Principal" field in the policy. If it is set to "*", this means that the topic allows global subscribe, which is a misconfiguration. The "Principal" field should be set to specific AWS accounts or AWS services that need to publish or subscribe to the topic.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the SNS topics.
+
+2. Once the AWS CLI is set up, you can list all the SNS topics in your account using the following command:
+   ```
+   aws sns list-topics
+   ```
+   This command will return a list of all SNS topics in your account.
+
+3. For each SNS topic, you can get the topic attributes using the following command:
+   ```
+   aws sns get-topic-attributes --topic-arn <TopicARN>
+   ```
+   Replace `<TopicARN>` with the ARN of the SNS topic. This command will return the attributes of the specified SNS topic.
+
+4. Check the `Policy` attribute of each SNS topic. If the `Policy` attribute contains a statement with `"Effect": "Allow"` and `"Principal": "*"`, it means that the SNS topic allows global subscribe. This is a misconfiguration and should be fixed. You can use a Python script to automate this process. Here is a simple example:
+   ```python
+   import json
+   import subprocess
+
+   # Get the list of all SNS topics
+   topics = json.loads(subprocess.check_output(['aws', 'sns', 'list-topics']))['Topics']
+
+   for topic in topics:
+       # Get the attributes of each SNS topic
+       attributes = json.loads(subprocess.check_output(['aws', 'sns', 'get-topic-attributes', '--topic-arn', topic['TopicArn']]))['Attributes']
+       
+       # Check the policy of each SNS topic
+       policy = json.loads(attributes['Policy'])
+       for statement in policy['Statement']:
+           if statement['Effect'] == 'Allow' and statement['Principal'] == '*':
+               print(f"The SNS topic {topic['TopicArn']} allows global subscribe.")
+   ```
+   This script will print the ARNs of all SNS topics that allow global subscribe.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing Python scripts to interact with AWS, you need to install the necessary libraries. The primary library you need is boto3, which is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Set up AWS credentials: Boto3 needs your AWS credentials (access key and secret access key) to interact with AWS services. You can set them up in your environment variables:
+
+   ```python
+   import os
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_access_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>